### ISSUE TITLE
Usage examples for generated client.

### ISSUE THREAD

**Issue by aktary** [contiamo/restful-react#267]
**Describe the bug**
The README says `You can see a concrete usage inside the examples folder`, but there's no example of *actual usage*.  I see a config, a yaml, and a fetchers.ts, but no examples of how you actually use the API that's generated.  I've been able to generate my own `fetchers.ts` from my `swagger.json`, but have no idea how to use it.  As I'm coming from old-school class-based React and fairly new to hooks, it's really confusing.  

**To Reproduce**
Look at the examples folder.

**Expected behavior**
There's an example of _usage_ of the generated custom hooks.
**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by aktary** [comment]
Ok... update.  I've figured out that I can follow the examples for useGet() (and, presumably useMutuate()) and they'll generally apply to my generated hooks.  Closing for now.

**Comment by aktary** [comment]
Yeah... I'm still lost. Is it possible to use the generated code outside of a tsx context?  If I've got a simple TS module that does some network requests, can it use the custom generated APIs?  If so, how?  An example of _an implementation_ of the custom generated code would be soooo helpful.

**Comment by fabien0102** [comment]
Sorry, I didn't see your message before.

All the examples are in the `package.json`

https://github.com/contiamo/restful-react/blob/61604ee50626b3c910224225cf4dd92cf4b260b0/package.json#L50-L54

You can clone the repo and try them ;) (just replace `node dist/bin/restful-react.js` by `restful-react` when you want to run this in your project)

The generator only generate typescript files, but should be able to compile them in js and have the autocompletion.

The usage is quite simple, let me try to resume the flow:
- you have a swagger file with a `operationId: listPets` https://github.com/contiamo/restful-react/blob/61604ee50626b3c910224225cf4dd92cf4b260b0/examples/petstore.yaml#L10-L41
- you run the generator with your swagger as input and generated file (tsx) as output (`example:file` in the package.json of this repo)
- you have `ListPet` and `useListPet` components is the generated file (with all the related types)
```tsx
/* Generated by restful-react */

import React from "react";
import { Get, GetProps, useGet, UseGetProps } from "restful-react";

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export interface Pet {
  id: number;
  name: string;
  tag?: string;
}

export type Pets = Pet[];

export interface Error {
  code: number;
  message: string;
}

export interface ListPetsQueryParams {
  /**
   * How many items to return at one time (max 100)
   */
  limit?: number;
}

export type ListPetsProps = Omit<GetProps<Pets, Error, ListPetsQueryParams, void>, "path">;

/**
 * List all pets
 */
export const ListPets = (props: ListPetsProps) => (
  <Get<Pets, Error, ListPetsQueryParams, void>
    path={`/pets`}
    {...props}
  />
);

export type UseListPetsProps = Omit<UseGetProps<Pets, Error, ListPetsQueryParams, void>, "path">;

/**
 * List all pets
 */
export const useListPets = (props: UseListPetsProps) => useGet<Pets, Error, ListPetsQueryParams, void>(`/pets`, props);
```

```tsx
import { useListPet } from "./petStore"

const MyComp = () => {
 const {data, loading, error} = useListPet() 

  return <div>{loading ? "loading…" : data.map(pet => pet.name).join(" ")</div>
}
```
